lib/test/t_LeastSquaresEquationsSolver_std.cxx

```diff
@@ -0,0 +1,56 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The test file of class LeastSquaresEquationsSolver for standard methods
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/OT.hxx"
+#include "openturns/OTtestcode.hxx"
+
+using namespace OT;
+using namespace OT::Test;
+
+int main(int, char *[])
+{
+  TESTPREAMBLE;
+  OStream fullprint(std::cout);
+
+  try
+  {
+    /** Analytical construction */
+    Description input = {"x","y"};
+    Description formulas = {"y*x-sin(2*x)","1 + cos(y) + x"};
+
+    SymbolicFunction analytical(input, formulas);
+    LeastSquaresEquationsSolver myAlgo;
+    myAlgo.setResidualError(1e-5);
+    fullprint << "myAlgo=" << myAlgo << std::endl;
+    Point startingPoint = {2.0,1.0};
+    OSS oss;
+    oss.setPrecision(5);
+    fullprint << "x=" << String(oss << myAlgo.solve(analytical, startingPoint)) << std::endl;
```

Comment:
Unit tests based on expected outputs tend to be difficult to maintain. Please use assert features for ottest instead. Does the exact solution of this problem is known? If not, I suggest to use [Rosenbrock's test problem](https://en.wikipedia.org/wiki/Rosenbrock_function) which has exactly the same mathematical structure, i.e. a system of two equations solved by least squares (plus, the exact solution is known).